content/components/text-input.mdx

```diff
@@ -149,15 +149,61 @@ When a text input is processing some action like performing a search, a loading
+
+If the text input is set to `disabled`, the rendered `<input>` includes the `disabled` attribute.
+
+The colors used to visually denote the validation state of the input meet minimum color contrast requirements.
+
+## Implementation requirements
+
+When using Octicons for leading and trailing visuals, note that icons don't have any text alternative. They are purely visual, and not conveyed to screen readers. Don't rely on these icons alone to convey meaning – make sure that the text label of the button provides sufficient meaning/context on its own.
+
+In the React version, when setting the `loading` property, the component will include a visual spinner. Note that this spinner is not conveyed to screen readers. You will need to use additional methods (such as a visually hidden live region notification) to convey this state to screen reader users. See the [documentation for loading](https://primer.style/ui-patterns/loading) for further details.
+
+While the Rails version of the component allows for the visible label of the text input to be hidden using `visually_hide_label`, this should only be done if there is an equivalent visible text that acts as the label. In that case, make sure that the actual label / accessible name of the input matches, or is at least contained within, the visible text acting as the label.
+
+## How to test the component
+
+### Integration tests
+
+- The text input has a sufficiently descriptive visible label, and does not rely on the use of placeholder text alone to act as the visible label
+- If the text input includes icons for its leading or trailing visuals, the input's purpose is clear even without the icons (as these are only decorative and are not conveyed to screen reader users)
+- In the React version, when using the `loading` state, this state is conveyed appropriately to screen reader users
+
+### Component tests
+
+- The component renders a `<label>` element, containing the `name`, and an `<input>`; the `<label for="…">` attribute must point to the `id` of the `<input>`
+- If the component is set to `disabled`, the rendered `<input>` includes the `disabled` HTML attribute
+- If the component has a `placeholder` attribute, the rendered `<input>` includes the `placeholder` HTML attribute
+- In the Rails implementation, if the component has a `caption`, the caption is rendered as an additional container underneath the `<input>`, and the `<input>` has an `aria-describedby="…"` pointing to the `id` attribute of the container
+- In the Rails implementation, if the component has a `validationMessage`, the message is rendered as an additional container underneath the `<input>`, and the `<input>` has an `aria-describedby="…"` pointing to the `id` attribute of the container
+- In the Rails implementation, if the component is set to `invalid`, the rendered `<input>` includes the `aria-invalid="true"` attribute
```

Comment:
Again, we go into details about these Rails features, but not the React features.

Reply:
Nevermind, we address this below